src/main/java/com/customer/controller/CreditCardController.java

```diff
@@ -0,0 +1,48 @@
+package com.customer.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.customer.exception.ForbiddenException;
+import com.customer.exception.InvalidRequest;
+import com.customer.model.CreditCard;
+import com.customer.service.CreditCardServiceImpl;
+
+@RestController
+@RequestMapping("/api")
+public class CreditCardController {
+	
+	private CreditCardServiceImpl creditCardServiceImpl;
```

Comment:
why have the implementation here. always autowire to the base class (I will ask you question why do we do that)

Reply:
Done , we should use the base class object to access the child class as with the same object we can access both he classes objects.And if there will be any chng in base class then automatically child classes will get update.